llpc/context/llpcContext.cpp

```diff
@@ -85,13 +85,24 @@ LgcContext *Context::getLgcContext() {
     // First time: Create the LgcContext.
     std::string gpuName = LgcContext::getGpuNameString(m_gfxIp.major, m_gfxIp.minor, m_gfxIp.stepping);
     m_builderContext.reset(
-        LgcContext::create(*this, gpuName, PAL_CLIENT_INTERFACE_MAJOR_VERSION, CodeGenOpt::Level::Default));
+        LgcContext::create(*this, gpuName, PAL_CLIENT_INTERFACE_MAJOR_VERSION, getOptimizationLevel()));
     if (!m_builderContext)
       report_fatal_error(Twine("Unknown target '") + Twine(gpuName) + Twine("'"));
   }
   return &*m_builderContext;
 }
 
+// =====================================================================================================================
+// Returns the optimization level
+CodeGenOpt::Level Context::getOptimizationLevel() const {
+#if LLPC_CLIENT_INTERFACE_MAJOR_VERSION > 52
+  if (getPipelineContext()->getPipelineOptions()->optimizationLevel <= 3)
+    return static_cast<CodeGenOpt::Level>(getPipelineContext()->getPipelineOptions()->optimizationLevel);
```

Comment:
nit: maybe hoist `getPipelineContext()->getPipelineOptions()->optimizationLevel()` to a local variable?